x/sequencer/keeper/rotation.go

```diff
@@ -95,3 +105,44 @@ func (k Keeper) OnProposerLastBlock(ctx sdk.Context, proposer types.Sequencer) e
 	)
 	return nil
 }
+
+// setSuccessorForRollapp will assign a successor. It won't replace an existing one.
+// It will prioritize non sentinel
+func (k Keeper) setSuccessorForRollapp(ctx sdk.Context, rollapp string) error {
+	successor := k.GetSuccessor(ctx, rollapp)
+	if !successor.Sentinel() {
+		// a valid successor is already set so there's no need to do anything
+		return nil
+	}
```

Comment:
as to my other comment of the `RecoverFromHaltIfNeeded` would sugget to remove outside the function.